packages/roosterjs-editor-dom/lib/selection/createRange.ts

```diff
@@ -44,3 +103,43 @@ function getFocusablePosition(position: NodePosition) {
         ? new Position(position.node, position.isAtEnd ? PositionType.After : PositionType.Before)
         : position;
 }
+
+function isNodePosition(arg: any): arg is NodePosition {
+    return arg && arg.node;
+}
+
+function getPositionFromPath(node: Node, path: number[]): NodePosition {
+    if (!node || !path) {
+        return null;
+    }
+
+    // Iterate with a for loop to avoid mutating the passed in element path stack
+    // or needing to copy it.
+    let offset: number;
+
+    for (let i = 0; i < path.length; i++) {
+        offset = path[i];
+        if (
+            i < path.length - 1 &&
+            node &&
+            node.nodeType == NodeType.Element &&
+            node.childNodes.length > offset
+        ) {
+            node = node.childNodes[offset];
+        } else {
+            break;

```

Comment:
If you always break out of the for loop once the conditional is not true, doesn't that mean you only ever check the index of 0? Why do you need a loop in this case?

Reply:
This is existing code, moved from getSelectionPath.ts. It has been there for a while and run perfectly. So I believe we don't need to change it.